jubilant/_all_any.py

```diff
@@ -38,8 +50,10 @@ def all_error(status: Status, apps: Iterable[str] | None = None) -> bool:
     return _all_statuses_are('error', status, apps)
 
 
-def all_maintenance(status: Status, apps: Iterable[str] | None = None) -> bool:
-    """Report whether all applications and units in *status* are in "maintenance" status.
+def all_maintenance(status: Status, *apps: str) -> bool:
+    """Report whether all apps and units in *status* (or in *apps* if provided) are "maintenance".
```

Comment:
Minor: the headline is getting a little long.

Reply:
Maybe: 

Additionally, in the arguments, consider adding some variation of: that if apps are provided, they must be present in status, or the return is False.

Reply:
Second comment -- I've added a phrase to each of the `all_*` docs "args" help text.